### ISSUE TITLE
Review dataset.py

### ISSUE THREAD

**Issue by XDynames** [AdrianOrenstein/deeplabv3plus-pytorch#4]
Target encoding likely needs to be re-examined for all the dataset classes in dataset.py. Primarily because lightning's IoU doesn't let us provide a custom ignore label which will need to be handled in both IoU calculation and cross-entropy loss.

Additionally cityscapes is loading targets with a much larger set of class ID values than expected. All classes should be checked though to ensure correct loading.
